src/injected/components/details-dialog.tsx

```diff
@@ -167,6 +174,10 @@ export class DetailsDialog extends React.Component<DetailsDialogProps, DetailsDi
             shouldShowInspectButtonMessage: () =>
                 this.props.dialogHandler.shouldShowInspectButtonMessage(this),
             userConfigurationStoreData: this.state.userConfigurationStoreData,
+            hasSecureTargetPage: this.props.dialogHandler.isTargetPageOriginSecure(),
+            shouldShowInsecureOriginPageMessage: () =>
```

Comment:
You've done a good job of using a consistent pattern with what the existing `shouldShowInspectButtonMessage` is doing, but unfortunately I think the existing code has set a bad example for you to follow. I think it would be better to pass `this.state.showInsecureOriginPageMessage` as a value here, rather than passing a prop which is a function that returns that value.

React chooses whether to re-render a child component as part of re-rendering a parent component based on whether the props passed to the child have changed. However, by passing a prop of the form `() => something`, React is going to see the prop changing with every parent render (since the parent is creating a new lambda function to pass on every render). That's not ideal; it would be better to pass the value from the state as the prop, so that React can detect whether the state value has changed from render-to-render, and can avoid re-rendering the child component unnecessarily if that value hasn't changed.

Reply:
Changing this won't have a direct impact on its own without changing the existing issues in the similar props; if any single prop has this issue, the whole child component will have this issue. It's not necessary to fix the old code as part of this PR; it would probably be better to do that in its own separate PR. But as you've seen, it's really easy for new code in the future to copy from what old code is doing, so I think it would be nice to improve this for the new code even in the absence of fixing the old code.